appdaemontestframework/assert_that.py

```diff
@@ -120,6 +124,23 @@ def called_with(self, **kwargs):
         self.hass_functions['call_service'].assert_any_call(
             service_full_name, **kwargs)
 
+    def set_to_option(self, newvalue, **service_specific_parameters):
```

Comment:
@FlorianKempenich I think I need some help here. I don't understand how it works. Never used lambdas and all this part seems to obscure to me.

Reply:
Yes I agree that part is a bit obscure. I tried to make it as readable as possible, but even for someone familiar w/ lambda it's not the most self-explanatory code . . . 

The part with the lambda is for the turned on/off assertions. For **turning on/off** a light, there are 2 ways to do so, from the automation we can call `self.turn_on(ENTITY)` **or** `self.call_service('turn_on', {'entity_id': ENTITY})`. What the helper does when we call `assert_that(ENTITY).was.turned_on()`: it checks both (`turn_on` and `call_service`) and see if at least one was called. If none was called, it'll collect errors from both and merge them into 1. `_capture_assert_failure_exception` runs the lambda and capture any failure or `None` if there was no failure. Then we check in `if service_not_called and turn_on_helper_not_called:` if any of the 2 calls returned an error,, and if so we raise a merged error 
```
raise EitherOrAssertionError(
                service_not_called, turn_on_helper_not_called)
```

In the case of `set_option`, if there is a way to call `set_option` via a service call, then we should be able to replicate a similar behavior indeed. I'll now look into why your test is failing :)